cw/capture.py

```diff
@@ -991,6 +1008,235 @@ def kmac_fvsr_key_batch(ctx: typer.Context,
     capture_end(ctx.obj.cfg)
 
 
+def capture_otbn_vertical(ot, ktp, fw_bin, pll_frequency, capture_cfg):
+    """Capture traces for ECDSA P256/P384 secret key generation.
+
+    Uses a fixed seed and generates several random masks. For the corresponding
+    driver, check:
+    <opentitan_repo_root>/sw/device/sca/ecc256_keygen_serial.c
+
+    Args:
+      ot: Initialized OpenTitan target.
+      ktp: Key and plaintext generator.
+      fw_bin: Firmware binary.
+      pll_frequency: Output frequency of the FPGA PLL.
+      capture_cfg: Capture configuration from the yaml file
+    """
+
+    # We need an intuitive, but non default behavior for the kpt.next() iterator.
+    # For backwards compatibility this must be set in the capture config file.
+    # This is a workaroung for https://github.com/lowRISC/ot-sca/issues/116
+    if "use_fixed_key_iter" not in capture_cfg:
+        raise RuntimeError('use_fixed_key_iter not set!')
+    if capture_cfg["use_fixed_key_iter"] is not False:
+        raise RuntimeError('use_fixed_key_iter must be set to false!')
+
+    # Currently, we need to reprogram the firmware before every sign operation
+    # TODO: Investigate if this is needed.
+    reset_firmware = False
+
+    # OTBN operations are long. CW-Husky can store only 131070 samples
+    # in the non-stream mode.
+    fifo_size = 131070
+    if capture_cfg["num_samples"] > fifo_size:
+        raise RuntimeError('Current setup only supports up to 130k samples')
+
+    # Be sure we don't use the stream mode
+    if ot.scope._is_husky:
+        ot.scope.adc.stream_mode = False
+        ot.scope.adc.bits_per_sample = 12
+        ot.scope.adc.samples = capture_cfg["num_samples"]
+    else:
+        # TODO: Add cw-lite support
```

Comment:
Hm, I am not sure I understand your comment. I think we don't want to have a TODO to Add CW-Lite support, because we're never going to add it but to error out if CW-Lite is detected is the best thing to do here.